src/nvim/lua/converter.c

```diff
@@ -703,14 +703,10 @@ void nlua_push_Boolean(lua_State *lstate, const Boolean b, int flags)
 void nlua_push_Dictionary(lua_State *lstate, const Dictionary dict, int flags)
   FUNC_ATTR_NONNULL_ALL
 {
-  if (dict.size == 0 && (flags & kNluaPushSpecial)) {
-    nlua_create_typed_table(lstate, 0, 0, kObjectTypeDictionary);
-  } else {
-    lua_createtable(lstate, 0, (int)dict.size);
-    if (dict.size == 0 && !(flags & kNluaPushSpecial)) {
-      nlua_pushref(lstate, nlua_global_refs->empty_dict_ref);
-      lua_setmetatable(lstate, -2);
-    }
+  lua_createtable(lstate, 0, (int)dict.size);
+  if (dict.size == 0 && !(flags & kNluaPushSpecial)) {
```

Comment:
if we do this, we should also remove the `!(flags & kNluaPushSpecial)` condition here, so that the (less intrusive) metatable tag is used instead.